File: techadvisor/src/main/java/com/example/techadvisor/dto/ChatMessageDTO.java
package com.example.techadvisor.dto;

import lombok.Data;

import java.util.List;

@Data
public class ChatMessageDTO {
    private Long id;
    private String senderType;
    private String content;
    private String createdAt;
    private List<ProductDTO> suggestedProducts;
}

File: GraduateProject/app/src/main/java/com/example/graduateproject/data/remote/dto/ChatMessageDto.kt
package com.example.graduateproject.data.remote.dto

import com.example.graduateproject.domain.model.Product
import kotlinx.serialization.Serializable

@Serializable
data class ChatMessageDTO(
    val id: Long,
    val senderType: String, // "USER" hoặc "AI"
    val content: String,
    val createdAt: String?,
    val suggestedProducts: List<Product>? = null
)